monai/visualize/occlusion_sensitivity.py

```diff
@@ -301,29 +277,73 @@ def __call__(
                 * The most probable class when the corresponding part of the image is occluded (``argmax(dim=-1)``).
             Both images will be cropped if a bounding box used, but voxel sizes will always match the input.
         """
+        if x.shape[0] > 1:
+            raise ValueError("Expected batch size of 1.")
+
+        sd = x.ndim - 2
+        mask_size = ensure_tuple_rep(self.mask_size, sd)
+
+        # get the meshgrid (so that sliding_window_inference can tell us which bit to occlude)
+        grid: MetaTensor = MetaTensor(
```

Comment:
just another thought that by defining a binary mask to indicate the area to occlude, along with this `grid` variable, it seems it'll be possible to make occlusion of arbitrary shapes (not necessarily a set of rectangular bounding boxes).

Reply:
Yes! The Gaussian is already arguably circular/spherical as the voxels are affected radially from the centre of the occluded region.